### DISCUSSION TITLE
Users are unable to create a new account using an oAuth provider (slack)

### DISCUSSION THREAD

**Discussion by javruben** [requarks/wiki#3343] [Help / Questions]
**Describe the bug**
The only way to see the "create account" link that allows account creation for the oAuth provider (in this case slack) is to click on the logo of that provider. However, when a user clicks on that logo it immediately tries to log in with that provider, which ultimately fails because the user doesn't exist. A catch-22. Clicking on the create account link only creates local accounts.

**To Reproduce**
1. Go to 'https://domain/login'
2. Click on 'oAuth provider logo'
3. Navigates away to log in.

**Expected behavior**
Users want to create an account using an oAuth provider

**Host Info (please complete the following information):**
 - OS: [e.g. Ubuntu]
 - Wiki.js version: [e.g. 2.0.0-beta.208]
 - Database engine: [e.g. postgres 10]

**Comment by NGPixel** [comment]
Did you enable the "Allow Self-Registration" option in the Slack configuration in the admin area?

**Comment by javruben** [comment]
Yes, first I only did it for slack. But then it was impossible to get to the register link. So then I enabled it for local. But then I noticed that clicking on that link only shows the page to add a user for local not for slack.

**Comment by NGPixel** [comment]
There's no registration link for OAuth providers, that's what the Allow Self-Registration option is for. Once the login succeeds with Slack, the account will be automatically created in Wiki.js.

**Comment by javruben** [comment]
Ok, then the issue is perhaps different. I get to this:
<img width="504" alt="Screen Shot 2019-07-11 at 3 08 19 PM" src="https://user-images.githubusercontent.com/248865/61029526-d05f6480-a3ed-11e9-983b-4bf2df86de8d.png">

And it then redirects me to this:
<img width="503" alt="Screen Shot 2019-07-11 at 3 08 32 PM" src="https://user-images.githubusercontent.com/248865/61029550-df461700-a3ed-11e9-8887-43c3ab087798.png">

Could this be a configuration error on my side?

**Comment by NGPixel** [comment]
You need to assign the user to a group (which you can do automatically by configuring the Allow Self-Registration option).

That group must have access to that page. Make sure the `read:pages` global permission is activated and that a Page Rule exists for that path.

**Comment by javruben** [comment]
I did indeed configure a group.
<img width="428" alt="Screen Shot 2019-07-11 at 3 12 51 PM" src="https://user-images.githubusercontent.com/248865/61029783-63989a00-a3ee-11e9-8d26-b13e0efbe4fb.png">

The page (I assume) is just the home page (currently that's the only page there is).

**Comment by javruben** [comment]
I didnt create a page rule... let me check.

**Comment by javruben** [comment]
Ok I just changed it to the guest group, instead of the group I manually created. I get the same error.

**Comment by javruben** [comment]
Also note; no users are actually created when looking at the users tab in the admin area

**Comment by lararosekelley** [comment]
FWIW, seeing this now on that latest release candidate with the Auth0 provider. Configured it so self-registration is available, created the app in Auth0, and can see in the Auth0 dashboard that logins are successful.

However, on the wiki side, I get an "Unauthorized" page and the server response is a 500. This is for users that exist in Auth0 trying to log into the Wiki.js site. Happy to share more details, but unforunately nothing interesting in the console logs / HTTP response and I'm not sure where to look on the server for more detailed logs.
